src/data/equiv/fin.lean

```diff
@@ -149,6 +169,63 @@ def sum_fin_sum_equiv : fin m ⊕ fin n ≃ fin (m + n) :=
     { dsimp, rw [dif_neg H], simp [fin.ext_iff, nat.add_sub_of_le (le_of_not_gt H)] }
   end }
 
+/-- The equivalence between `fin (m + n)` and `fin (n + m)` which rotates by `n`. -/
+def fin_add_flip : fin (m + n) ≃ fin (n + m) :=
+  (sum_fin_sum_equiv.symm.trans (equiv.sum_comm _ _)).trans sum_fin_sum_equiv
```

Comment:
Can't you use `fin_congr (add_comm _ _)` here?

Reply:
But then it won't have the correct behaviour: this would produce "the identity".

Reply:
Ignore me!